packages/rum/src/getActionNameFromElement.ts

```diff
@@ -0,0 +1,176 @@
+      return getTextualContent(element)
+    }
+  },
+  // aria-label attribute value
+  (element) => element.getAttribute('aria-label'),
+  // associated element text designated by the aria-labelledby attribute
+  (element) => {
+    const labelledByAttribute = element.getAttribute('aria-labelledby')
+    if (labelledByAttribute) {
+      return labelledByAttribute
+        .split(/\s+/)
+        .map((id) => getElementById(element, id))
+        .filter((label): label is HTMLElement => Boolean(label))
+        .map(getTextualContent)
+        .join(' ')
+    }
+  },
+  // alt attribute value
+  (element) => element.getAttribute('alt'),
+  // name attribute value
+  (element) => element.getAttribute('name'),
+  // title attribute value
+  (element) => element.getAttribute('title'),
+  // placeholder attribute value
+  (element) => element.getAttribute('placeholder'),
+  // SELECT first OPTION text
+  (element) => {
+    if ('options' in element && element.options.length > 0) {
+      return getTextualContent(element.options[0])
+    }
+  },
+]
+
+const fallbackGetters: NameGetter[] = [
+  // element text
+  (element) => {
+    return getTextualContent(element)
+  },
+]
+
+function getActionNameFromElementForGetters(targetElement: Element, getters: NameGetter[]) {
+  let element: Element | null = targetElement
+  let recursionCounter = 0
+  while (
+    recursionCounter <= 10 &&
+    element &&
+    element.nodeName !== 'BODY' &&
+    element.nodeName !== 'HTML' &&
+    element.nodeName !== 'HEAD'
+  ) {
+    for (const getter of getters) {
+      const name = getter(element)
+      if (typeof name === 'string') {
+        const trimedName = name.trim()
+        if (trimedName) {
+          return truncate(normalizeWhitespace(trimedName))
+        }
+      }
+    }
+    // Don't consider parents of FORM elements
+    if (element.nodeName === 'FORM') {
+      break
+    }
+    element = element.parentElement
+    recursionCounter += 1
+  }
+}
+
+function normalizeWhitespace(s: string) {
+  return s.replace(/\s+/g, ' ')
+}
+
+function truncate(s: string) {
+  return s.length > 100 ? `${s.slice(0, 100)} [...]` : s
+}
+
+function getElementById(refElement: Element, id: string) {
+  // tslint:disable-next-line: no-null-keyword
+  return refElement.ownerDocument ? refElement.ownerDocument.getElementById(id) : null
+}
+
+function getTextualContent(element: Element | HTMLElement) {
+  if ((element as HTMLElement).isContentEditable) {
+    return
+  }
+
+  if ('innerText' in element) {
+    let text = element.innerText
+    if (!supportsInnerTextScriptAndStyleRemoval()) {
+      // remove the inner text of SCRIPT and STYLES from the result. This is a bit dirty, but should
+      // be relatively fast and work in most cases.
+      const elementsToRemove: NodeListOf<HTMLElement> = element.querySelectorAll('script, style')
+      // tslint:disable-next-line: prefer-for-of
+      for (let i = 0; i < elementsToRemove.length; i += 1) {
```

Comment:
according to MDN for-of should work correctly on node list, typing does not seem to allow that though.
it could worth to check if it is supported by newer ts version but not important now

Reply:
It appears to be working on two conditions:
* use the DOM.Iterable `lib`
* and use ES6+ as a `target` or the `downlevelIteration` option
I'll keep the traditional `for` loop here for now.